src/engine/QueryPlanner.cpp

```diff
@@ -737,11 +840,11 @@ QueryPlanner::TripleGraph QueryPlanner::createTripleGraph(
 // _____________________________________________________________________________
 vector<QueryPlanner::SubtreePlan> QueryPlanner::seedWithScansAndText(
     const QueryPlanner::TripleGraph& tg,
-    const vector<QueryPlanner::SubtreePlan*>& children) const {
+    const vector<const QueryPlanner::SubtreePlan*>& children) {
```

Comment:
Currently we don't use `shared_ptr` as the data is stored in the `childPlanStorage` and `subqueryPlans` vectors which are guaranteed to have the required lifetime to ensure that the data is never used after deletion. Unless we want to use these `SubtreePlans` somewhere else `shared_ptr` seems like an unecessary overhead to me (although it might not be noticeable in the end, given the small size of the `children` vector).